web/react/components/edit_channel_modal.jsx

```diff
@@ -55,9 +55,13 @@ export default class EditChannelModal extends React.Component {
         const button = e.relatedTarget;
         this.setState({header: $(button).attr('data-header'), title: $(button).attr('data-title'), channelId: $(button).attr('data-channelid'), serverError: ''});
     }
+    handleShown() {
+        $('#edit_channel #edit_header').focus();
+    }
     componentDidMount() {
         $(ReactDOM.findDOMNode(this.refs.modal)).on('show.bs.modal', this.onShow);
         $(ReactDOM.findDOMNode(this.refs.modal)).on('hidden.bs.modal', this.handleClose);
+        $(ReactDOM.findDOMNode(this.refs.modal)).on('shown.bs.modal', this.handleShown);
```

Comment:
I think you need `this.handleShown = this. handleShown.bind(this);` in the constructor